tests/Microsoft.NET.SDK.ToolInstaller/Microsoft.NET.SDK.ToolInstaller.Tests.ps1

```diff
@@ -0,0 +1,103 @@
+#>
+
+BeforeAll {
+    Install-Module -Name PSDesiredStateConfiguration -Force -SkipPublisherCheck
+    Import-Module Microsoft.NET.SDK.ToolInstaller
+}
+
+Describe 'List available DSC resources' {
+    It 'Shows DSC Resources' {
+        $expectedDSCResources = "NETSDKToolInstaller"
+        $availableDSCResources = (Get-DscResource -Module Microsoft.NET.SDK.ToolInstaller).Name
+        $availableDSCResources.count | Should -Be 1
+        $availableDSCResources | Where-Object { $expectedDSCResources -notcontains $_ } | Should -BeNullOrEmpty -ErrorAction Stop
+    }
+}
+
+Describe 'DSC operation capabilities' {
+    It 'Test for empty package.' -Skip:(!$IsWindows) {
+        $parameters = @{
+            PackageId = 'dotnet'
+        }
+        
+        $testResult = Invoke-DscResource -Name NETSDKToolInstaller -ModuleName Microsoft.NET.SDK.ToolInstaller -Method Test -Property $parameters -Verbose -Debug
+        $testResult.InDesiredState | Should -Be $false # because is empty
+    }
+
+    It 'Sets desired package' -Skip:(!$IsWindows) {
+        $parameters = @{
+            PackageId = 'GitVersion.Tool'
+        }
+        
+        Invoke-DscResource -Name NETSDKToolInstaller -ModuleName Microsoft.NET.SDK.ToolInstaller -Method Set -Property $parameters
+     
+        $finalState = Invoke-DscResource -Name NETSDKToolInstaller -ModuleName Microsoft.NET.SDK.ToolInstaller -Method Get -Property $parameters
+        $finalState.PackageId | Should -Be $parameters.PackageId
+        $finalState.Exist | Should -BeTrue
+     }
+
+     It 'Sets desired package with prerelease' -Skip:(!$IsWindows) {
+        $parameters = @{
+            PackageId = 'dotnet-ef'
+            PreRelease = $true
+        }
+        
+        Invoke-DscResource -Name NETSDKToolInstaller -ModuleName Microsoft.NET.SDK.ToolInstaller -Method Set -Property $parameters
+     
+        $finalState = Invoke-DscResource -Name NETSDKToolInstaller -ModuleName Microsoft.NET.SDK.ToolInstaller -Method Get -Property $parameters
+        $finalState.PackageId | Should -Be $parameters.PackageId
+        $finalState.PreRelease | Should -BeTrue
+     }
+
+     It 'Sets desired package with version' -Skip:(!$IsWindows) {
+        $parameters = @{
+            PackageId = 'dotnet-reportgenerator-globaltool'
+            Version = '5.3.10'
+        }
+        
+        Invoke-DscResource -Name NETSDKToolInstaller -ModuleName Microsoft.NET.SDK.ToolInstaller -Method Set -Property $parameters
+     
+        $finalState = Invoke-DscResource -Name NETSDKToolInstaller -ModuleName Microsoft.NET.SDK.ToolInstaller -Method Get -Property $parameters
+        $finalState.PackageId | Should -Be $parameters.PackageId
+        $finalState.Version | Should -Be $parameters.Version
+     }
+
+     It 'Sets desired package with prerelease version' -Skip:(!$IsWindows) {
+        $parameters = @{
+            PackageId = 'PowerShell'
+            Version = '7.2.0-preview.5'
```

Comment:
why this version of powershell?

Reply:
If I'm not mistaken, the logic of the CLI to install packages is as followed:

1. Whenever a preview version is higher than the official version, you require the `--prerelease` parameter to be added without _version_
2. When the official version is higher than a preview version, you use the `--version` parameter to indicate you want to install a prerelease version that is older. Because you want to test for this scenario, I have included an older version of PowerShell. See list:

![image](https://github.com/user-attachments/assets/6250250c-13b1-43de-a2bd-bb03406fc4b5)